sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/exemplar/ExemplarSampler.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.exemplar;
+
+import com.google.auto.value.AutoValue;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * An interface that provides the two configuration components to exemplar sampling.
+ *
+ * <ul>
+ *   <li>(coming soon) A filter for which measurements should be sampled.
+ *   <li>A reservoir to store exemplars (and can do additional filtering)
+ * </ul>
+ */
+@AutoValue
+@Immutable
+public abstract class ExemplarSampler {
```

Comment:
Add package private constructor

Reply:
With the above suggestion, I guess this is `ExemplarSamplerConfig` or `ExemplarSamplerProvider`. Wonder if we actually need this class though instead of having all knobs on `ExemplarStrategy`

Reply:
This eventually get a measurement pre-filter, see: https://github.com/jsuereth/opentelemetry-java/blob/wip-metrics-api/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/exemplar/ExemplarSampler.java

Reply:
definitely needs a package private constructor, name change or not.

Reply:
Made this package private.  PTAL at #3599 if you want to see what `ExemplarSampler` will look like once completed.